pkg/component/kubernetes/controllermanager/controllermanager.go

```diff
@@ -302,10 +302,16 @@ func (k *kubeControllerManager) Deploy(ctx context.Context) error {
 	if _, err := controllerutils.GetAndCreateOrMergePatch(ctx, k.seedClient.Client(), service, func() error {
 		service.Labels = getLabels()
 
-		utilruntime.Must(gardenerutils.InjectNetworkPolicyAnnotationsForScrapeTargets(service, networkingv1.NetworkPolicyPort{
+		networkPolicyPort := networkingv1.NetworkPolicyPort{
 			Port:     utils.IntStrPtrFromInt32(port),
 			Protocol: ptr.To(corev1.ProtocolTCP),
-		}))
+		}
+
+		if k.values.NamePrefix != "" {
```

Comment:
Similar to the remark in `etcd.go`, i.e. it is fairly implicit and imposes some cognitive load on readers to deduce what this means.